Scripts/gen_FlatMuDaughter.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/bash
+#
+# create fcl for producing primaries from stopped particles
+# starting from a given primary that allows to set:
+# pdgID, startMom and endMom. 
+# Tested with the FlateMinus/Plus primaries
+#
+# this script requires mu2etools and dhtools be setup
+#
+# $1 is the primary
+# $2 is the production (ie MDC2020)
+# $3 is the stops production version
+# $4 is the output primary production version
+# $5 is the kind of input stops (Muminus, Muplus, IPAMuminus, IPAMuplus, or Cosmic)
+# $6 is the number of jobs
+# $7 is the number of events/job
+# $8 is the pdgId of the particle to generate
+# $9 is the startMom
+# $10 is the endMom
+# $11 is the name of the BField file
+#
+# example for running this script:
+#        source Production/Scripts/gen_FlatMuDaughter.sh FlatePlus MDC2020 k s Muminus 2 100 -11 30 100 "Offline/Mu2eG4/geom/bfgeom_DS70_no_tsu_ps_v01.txt"
+
+if [[ $# -lt 7 ]]; then
+  echo "Missing arguments, provided $# but there should be 7"
+  return 1
+fi
+primary=$1
+stopsconf=$2$3
+primaryconf=$2$4
+stype=$5
+njobs=$6
+eventsperjob=$7
+pdg=$8
+startMom=$9
+endMom=${10}
+bFieldFile=${11}
+
+echo PDG=$pdg
+echo startMom=$startMom
+echo endMom=$endMom
+echo bField=$bFieldFile
+
+if [[ "${stype}" == "Muminus" ]] ||  [[ "${stype}" == "Muplus" ]]; then
+  dataset=sim.mu2e.${stype}StopsCat.${stopsconf}.art
+  resampler=TargetStopResampler
+elif [[ "${stype}" == "Cosmic" ]]; then
+  dataset=sim.mu2e.${stype}DSStops${primary}.${stopsconf}.art   # should have Cat FIXME!
+  resampler=${stype}Resampler
+else
+  dataset=sim.mu2e.${stype}StopsCat.${stopsconf}.art
+  resampler=${stype}StopResampler
+fi
+
+samweb list-file-locations --schema=root --defname="$dataset"  | cut -f1 > Stops.txt
+# calucate the max skip from the dataset
+nfiles=`samCountFiles.sh $dataset`
+nevts=`samCountEvents.sh $dataset`
+let nskip=nevts/nfiles
+# write the template
+rm primary.fcl
+if [[ "${stype}" == "Cosmic" ]]; then
```

Comment:
Does this script make sense for cosmics?  I guess not, since those 'stops' are very different from target or IPA stops.  I suggest to remove this option.